RELEASE_GUIDE.md

```diff
@@ -44,48 +67,142 @@ considering what effects this may have:
 
 ## Making a release
 
-- Make a commit bumping versions. The following should be updated:
+A release is made by updating the versions within the repo via a PR. Once the PR is merged, a GitHub release is drafted, which triggers a CI build that should upload prebuilt binaries into that GitHub release page. (If it doesn't, you'll upload those manually).
+
+After that, the respective packages on Hackage and NPM will be uploaded/updated.
+
+### Set up a local branch for the PR
+
+```bash
+cd purescript
+git fetch upstream
+git checkout upstream/master
+git switch -c prepareNextRelease
+```
+
+### Make last changes and submit the PR
+
+```bash
+# 1. Regenerate the LICENSE file and commit it
+# See `./license-generator/` for details)
+make license-generator
+git add LICENSE
+git commit -m "Regenerated LICENSE file"
+
+# 2. Edit the changelog
+  # 2a. Create a new "Unreleased" section
+  # 2b. Rename the "Unreleased" section to
+  #   the version you will be releasing
+  # 2c. Edit the new version's content
+  #   until satisfied (e.g. grammer,
+  #   spelling, clarity, etc.)
+
+EDITOR="nano" # or your preferred editor
+$EDITOR CHANGELOG.md
+git add CHANGELOG.md
+git commit -m "Added new Unreleased section; finalized new version's section"
```

Comment:
I think "Finalizing CHANGELOG for {version}" might be a slightly clearer commit message, if we are going to suggest one here.